community/modules/scripts/spack-install/templates/spack_setup.yml.tpl

```diff
@@ -0,0 +1,77 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+- name: Install Application
+  hosts: localhost
+  vars:
+    app_name: ${app_name}
+    profile_script: ${profile_script}
+    install_dir: ${install_dir}
+    git_url: ${git_url}
+    git_ref: ${git_ref}
+    chmod_mode: ${chmod_mode}
+    chown_owner: ${chown_owner}
+    chgrp_group: ${chgrp_group}
+    finalize_setup_script: ${finalize_setup_script}
+  tasks:
+  - name: Print Application Name
+    ansible.builtin.debug:
+      msg: "Running installation for application: {{app_name}}"
+
+  - name: Add profile script for application
+    ansible.builtin.copy:
+      dest: /etc/profile.d/{{ app_name }}.sh
+      mode: '0644'
+      content: "{{ profile_script }}"
+
+  - name: Create parent of install directory
+    ansible.builtin.file:
+      path: "{{ install_dir | dirname }}"
+      state: directory
+
+  - name: Acquire lock
```

Comment:
I think the more natural way of doing this would be something like

```
- name: Create a directory if it does not exist
  ansible.builtin.file:
    register: lockdir
    path: /etc/some_directory
    state: directory
    mode: '0755'
```

(this apporach might also make it easier for it to be a lockfile vs dir)

Then block other things on `when: lockdir.changed`. It avoids setting `failed_when: false` which is a bit of a sledgehammer that can cause you to not see other problems.

I understand you may not want to make progress on other aspects of this functionality and this appears to work. If you wish, create a task to evaluate this alternative approach.

Reply:
As discussed offline, I have opened a task and will circle back in separate pr where I can perform dedicated testing.